src/screens/TaskDetailScreen.tsx

```diff
@@ -98,91 +63,84 @@ const TaskDetailScreen: React.FC<TaskDetailScreenProps> = ({ navigation, route }
   // Local state to manage the task
   const [task, setTask] = useState<Task>(initialTask);
 
-  // Animation values
-  const fadeAnim = useState(new Animated.Value(0))[0];
-
-  // Update local task state if initialTask changes
-  useEffect(() => {
-    setTask(initialTask);
-  }, [initialTask]);
-
-  // Start fade-in animation when component mounts
-  useEffect(() => {
-    Animated.timing(fadeAnim, {
-      toValue: 1,
-      duration: 300,
-      useNativeDriver: true
-    }).start();
-  }, []);
-
   /**
    * Toggles task status between 'pending' and 'completed'
    * Updates both local state and parent component
    * Provides an undo option via alert
    */
-  const handleToggleStatus = useCallback(() => {
-    // Store the old status for undo functionality
-    const oldStatus = task.status;
-
+  const handleToggleStatus = () => {
```

Comment:
Lines 71–112: The “Undo” logic captures task.status via closure, which can drift if you refactor. Instead, store the old status in a local variable:

`
const oldStatus = task.status;           // line 72
// ...
Alert.alert(..., [
  // …
  {
    text: 'Undo',
    onPress: () => {
      const reverted: Task = { ...task, status: oldStatus };
      setTask(reverted);
      onUpdate(reverted);
    },
  },
]);
`